File: app/src/main/java/cz/marvincz/rssnotifier/xml/XmlUriConverter.java
package cz.marvincz.rssnotifier.xml;

import android.net.Uri;

import org.apache.commons.lang3.reflect.TypeLiteral;
import org.apache.commons.lang3.reflect.Typed;

import androidx.annotation.Nullable;
import cz.marvincz.xmlpullparserconverter.AbstractXmlStringConverter;

public class XmlUriConverter extends AbstractXmlStringConverter<Uri> {
    @Override
    public Typed<Uri> getType() {
        return new TypeLiteral<Uri>() {};
    }

    @Nullable
    @Override
    protected Uri convertString(@Nullable String stringValue) {
        if (stringValue != null) {
            return Uri.parse(stringValue);
        }
        return null;
    }
}

File: app/src/main/java/cz/marvincz/rssnotifier/xml/XmlUriConverter.kt
package cz.marvincz.rssnotifier.xml

import android.net.Uri

import org.apache.commons.lang3.reflect.TypeLiteral
import org.apache.commons.lang3.reflect.Typed
import cz.marvincz.xmlpullparserconverter.AbstractXmlStringConverter

class XmlUriConverter : AbstractXmlStringConverter<Uri>() {
    override fun getType() = object : TypeLiteral<Uri>() {}

    override fun convertString(stringValue: String?): Uri? {
        return if (stringValue != null) {
            Uri.parse(stringValue)
        } else null
    }
}
